### PR TITLE
Reduced matching engine request batch size and added retry logic

### PR DESCRIPTION
**Pull request by KevinCLydon**
Match requests to matching engine will now be broken into chunks of 5 (determined by `GET_MATCHES_CHUNK_SIZE`) to be submitted sequentially.  Also added retry logic using the tenacity library.  The retry scheme right now uses 4 attempts per request, with exponential backoff for the wait time between requests.

### CODE REVIEW

File: src/casp/services/api/services/cell_operations_service.py
Comment (nmalfroy) [inline src/casp/services/api/services/cell_operations_service.py:22]: Should we make this configurable?

File: src/casp/services/api/services/cell_operations_service.py
Comment (nmalfroy) [inline src/casp/services/api/services/cell_operations_service.py:153]: Should we pull this into it's own method since it gets used in a few places.  Also maybe worth adding a unit test for said method?

**Review by nmalfroy — APPROVED** [review]
This looks great!  tenacity lib seems super useful.  A couple comments. I don't think they're blockers to merge but might be worth doing?  Your call

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py:22]: That's a good idea.  I could move it to the settings file and make it optionally configurable with an environment variable maybe?  I think that's the convention we're using currently.  Is that right @fedorgrab ?

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py:153]: Those are both good ideas.  I can make those changes

File: src/casp/services/api/services/cell_operations_service.py
Comment (fedorgrab) [inline src/casp/services/api/services/cell_operations_service.py:157]: Seems like this loop is identical to the one that's in `get_knn_matches` except for the method that's used to called inside of the retryer. 
Do you guys think it would be a good idea to wrap it up in some kind of a method that takes callback and its arguments (and maybe num_attempts)?

File: src/casp/services/api/services/cell_operations_service.py
Comment (fedorgrab) [inline src/casp/services/api/services/cell_operations_service.py:161]: I think making constants (or setting constants) from these numbers might be also a good idea).

File: src/casp/services/api/services/cell_operations_service.py
Comment (fedorgrab) [reply src/casp/services/api/services/cell_operations_service.py:110]: Also, I think we should increase number of attempts to a higher number, maybe like at least 10.

**Review by fedorgrab — CHANGES_REQUESTED** [review]
This is awesome! I did a stress test and the index showed up to 87 qps with almost no errors. 

I've left some small comments.

Also, could you please add the changes into the CHANGELOG.rst under existing 1.4.2 version? [Here's an example](https://github.com/cellarium-ai/cellarium-cas/blob/development/CHANGELOG.rst) of what sections can have the changelog file

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: Changes made!

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: Okay cool.  I added it to the settings file

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: It's super convenient

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: Moved it to settings and set the default to 10

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: I consolidated it into one function as you suggestion.  I changed two existing functions to wrappers that just pass the correct callback

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py]: Moved them all to settings

**Review by fedorgrab — APPROVED** [review]
Small cosmetic things, but please add updates to changelog before merging.

File: src/casp/services/api/services/cell_operations_service.py
Comment (KevinCLydon) [reply src/casp/services/api/services/cell_operations_service.py:184]: Thanks for catching this one too

**Comment by KevinCLydon** [comment]
> Small cosmetic things, but please add updates to changelog before merging.

@fedorgrab The previous entry in the changelog has a header with version number 1.4.2.  Should I use a new version number?

**Comment by fedorgrab** [comment]
> > Small cosmetic things, but please add updates to changelog before merging.
> 
> @fedorgrab The previous entry in the changelog has a header with version number 1.4.2. Should I use a new version number?

No, let's keep using 1.4.2, we well start a new version after the release. Version increments after each merge dev -> main, while dev aggregates all changelogs from feature branches.

**Comment by fedorgrab** [comment]
> > > > Small cosmetic things, but please add updates to changelog before merging.
> > > 
> > > 
> > > @fedorgrab The previous entry in the changelog has a header with version number 1.4.2. Should I use a new version number?
> > 
> > 
> > No, let's keep using 1.4.2, we well start a new version after the release. Version increments after each merge dev -> main (which is basically a release), while dev aggregates all changelogs from feature branches.
> 
> @fedorgrab Okay I've updated it. Can you take a quick look to make sure I did it correctly?

Looks Great!

**Review by fedorgrab — APPROVED** [review]

**Review by nmalfroy — APPROVED** [review]
looks great!

**Pull request merged**
